tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -2337,6 +2337,86 @@ public function provideIsSuperGlobalCases(): array
         return $cases;
     }
 
+    /**
+     * @dataProvider provideGetClassyModifiersCases
+     *
+     * @param array<string, int> $expectedModifiers
+     */
+    public function testGetClassyModifiers(array $expectedModifiers, int $index, string $source): void
+    {
+        $tokens = Tokens::fromCode($source);
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        static::assertSame($expectedModifiers, $tokensAnalyzer->getClassyModifiers($index));
+    }
+
+    public function provideGetClassyModifiersCases(): iterable
+    {
+        yield 'final' => [
+            ['final' => 1],
+            3,
+            '<?php final class Foo {}',
+        ];
+
+        yield 'abstract' => [
+            ['abstract' => 3],
+            5,
+            '<?php /* comment */ abstract class Foo {}',
+        ];
+    }
+
+    /**
+     * @requires PHP 8.2
+     *
+     * @dataProvider provideGetClassyModifiersOnPhp82Cases
+     *
+     * @param array<string, int> $expectedModifiers
+     */
+    public function testGetClassyModifiersOnPhp82(array $expectedModifiers, int $index, string $source): void
+    {
+        $tokens = Tokens::fromCode($source);
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        static::assertSame($expectedModifiers, $tokensAnalyzer->getClassyModifiers($index));
```

Comment:
Maybe we could reuse basic test instead of define the same logic again? This is here only because of `@requires PHP 8.2` so PHPUnit can ignore it for lower versions, but test's actual body is the same.

```suggestion
        $this->testGetClassyModifiers($expectedModifiers, $index, $source);
```

Too bad it's impossible to do [this](https://github.com/sebastianbergmann/phpunit/pull/2223#issuecomment-1249836668) 😉

Reply:
The repetition of code is a pain in utest, but than again it is only in the tests. If someone has a nice idea how to resolve it it would be great though.